lib/utils/docker.ts

```diff
@@ -110,7 +110,7 @@ export interface BuildOpts {
 	t?: string;
 }
 
-function parseBuildArgs(args: string[]): Dictionary<string> {
+export function parseBuildArgs(args: string[]): Dictionary<string> {
```

Comment:
I don't think I was the one who originally implemented this function, though I amended it at some point. If I was to reimplement it today (which I actually did before I remembered about it), 😁  I would have coded it without a regex, something like:
```ts
/**
 * Convert an array of variable assignment strings such as:
 *     [ "var1=value1", "var1=value2", "var+3=value3=something", "var4" ]
 * to a dictionary such as:
 *     { var1: "value2", "var+3": "value3=something", var4: "" }
 */
function varArrayToDictionary(varArray: string[]): Dictionary<string> {
	const dict: Dictionary<string> = {};
	for (const buildVar of varArray) {
		let key = buildVar;
		let value = '';
		const i = buildVar.indexOf('=');
		if (i > 0) {
			key = buildVar.substring(0, i);
			value = buildVar.substring(i + 1);
		}
		dict[key] = value;
	}
	return dict;
}
```

Reply:
Btw, the issue with `split('=')` is that it doesn't handle well the case where the variable value contains the equal character, which is the subject of bug #1705. Sample problematic command line:
```
balena push --env "POSTGRESQL_DSN=postgres://loraserver_ns:59XTH2995@postgresql/loraserver_ns?sslmode=disable" d19baeb.local
```
We should similarly handle such build arguments:
```
balena push --buildArg "POSTGRESQL_DSN=postgres://loraserver_ns:59XTH2995@postgresql/loraserver_ns?sslmode=disable" d19baeb.local
```

Reply:
Right, I kind of assumed this split on '=' would be a problem, but left it in for now as something to work from. Not sure how to handle it though off the top of my head.

Reply:
And as I'm guessing we can use `parseBuildArgs()` to convert build argument arrays to dictionaries as I mentioned earlier, refactoring it as a part of this PR wouldn't necessarily be out of scope.

Reply:
> [...] split on '=' would be a problem [...] Not sure how to handle it though
> we can use `parseBuildArgs()` to convert build argument arrays to dictionaries

But I have already done these two things in my commit! It handles the split on `=` and it uses `parseBuildArgs()` to convert build argument arrays to dictionaries. :-)